File: languages/js/src/entity/identity_v2/boolean_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

export class BooleanAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.BooleanAttributeDefinitionV2 {
    return identityEntitiesProto.BooleanAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.BooleanAttributeDefinitionV2
  ): BooleanAttributeDescriptor {
    return new BooleanAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}

File: languages/python/bloock/entity/identity_v2/boolean_attribute_descriptor.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute_descriptor import AttributeDescriptor


class BooleanAttributeDescriptor(AttributeDescriptor):
    @staticmethod
    def from_proto(a: proto.BooleanAttributeDefinitionV2) -> BooleanAttributeDescriptor:
        return BooleanAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
        )

    def to_proto(self) -> proto.BooleanAttributeDefinitionV2:
        return proto.BooleanAttributeDefinitionV2(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
        )
